drv/cosmo-seq-server/src/main.rs

```diff
@@ -414,6 +456,38 @@ impl ServerImpl {
                     return Err(CpuSeqError::A0Timeout);
                 }
 
+                let coretype0 = self.sys.gpio_read(SP5_TO_SP_CORETYPE0) != 0;
+                let coretype1 = self.sys.gpio_read(SP5_TO_SP_CORETYPE1) != 0;
+                let coretype2 = self.sys.gpio_read(SP5_TO_SP_CORETYPE2) != 0;
+                let sp5r1 = self.sys.gpio_read(SP5_TO_SP_SP5R1) != 0;
+                let sp5r2 = self.sys.gpio_read(SP5_TO_SP_SP5R2) != 0;
+                let sp5r3 = self.sys.gpio_read(SP5_TO_SP_SP5R3) != 0;
+                let sp5r4 = self.sys.gpio_read(SP5_TO_SP_SP5R4) != 0;
+
+                ringbuf_entry!(Trace::Coretype {
+                    coretype0,
+                    coretype1,
+                    coretype2,
+                    sp5r1,
+                    sp5r2,
+                    sp5r3,
+                    sp5r4
+                });
+
+                match (coretype0, coretype1, coretype2) {
+                    // These correspond to Type-2 and Type-3
+                    (true, false, true) | (true, false, false) => (),
+                    // Reject all other combos
+                    _ => return Err(CpuSeqError::UnrecognizedCPU),
```

Comment:
Should we return to A2 here (and below)?  It's weird to be in A0 with the HF mux still attached to the SP.

Reply:
You're right, I had somehow convinced myself we had error handling to do this already but we don't